cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -0,0 +1,851 @@
+pub fn connection_operations(
+    deps: DepsMut,
+    operations: Vec<ConnectionInput>,
+) -> Result<Response, ContractError> {
+    let response = Response::new();
+    for operation in operations {
+        match operation.operation {
+            Operation::Set => {
+                let channel_id =
+                    operation
+                        .channel_id
+                        .ok_or_else(|| ContractError::InvalidInput {
+                            message: "channel_id is required for set operation".to_string(),
+                        })?;
+                if CHAIN_TO_CHAIN_CHANNEL_MAP.has(
+                    deps.storage,
+                    (&operation.source_chain, &operation.destination_chain),
+                ) {
+                    return Err(ContractError::ChainToChainChannelLinkAlreadyExists {
+                        source_chain: operation.source_chain.clone(),
+                        destination_chain: operation.destination_chain.clone(),
+                    });
+                }
+                CHAIN_TO_CHAIN_CHANNEL_MAP.save(
+                    deps.storage,
+                    (&operation.source_chain, &operation.destination_chain),
+                    &channel_id,
+                )?;
+                if CHANNEL_ON_CHAIN_CHAIN_MAP
+                    .has(deps.storage, (&channel_id, &operation.source_chain))
+                {
+                    return Err(ContractError::ChannelToChainChainLinkAlreadyExists {
+                        channel_id,
+                        source_chain: operation.source_chain.clone(),
+                    });
+                }
+                CHANNEL_ON_CHAIN_CHAIN_MAP.save(
+                    deps.storage,
+                    (&channel_id, &operation.source_chain),
+                    &operation.destination_chain,
+                )?;
+                response.clone().add_attribute(
+                    "set_connection",
+                    format!("{}-{}", operation.source_chain, operation.destination_chain),
+                );
+            }
+            Operation::Change => {
+                let current_channel_id = CHAIN_TO_CHAIN_CHANNEL_MAP
+                    .load(
+                        deps.storage,
+                        (&operation.source_chain, &operation.destination_chain),
+                    )
+                    .map_err(|_| ContractError::ChainChannelLinkDoesNotExist {
+                        source_chain: operation.source_chain.clone(),
+                        destination_chain: operation.destination_chain.clone(),
+                    })?;
+                if let Some(new_channel_id) = operation.new_channel_id {
+                    CHAIN_TO_CHAIN_CHANNEL_MAP.save(
+                        deps.storage,
+                        (&operation.source_chain, &operation.destination_chain),
+                        &new_channel_id,
+                    )?;
+                    CHANNEL_ON_CHAIN_CHAIN_MAP
```

Comment:
No need to remove. Save already overrides.

Reply:
How is that possible if the key changes? For instance, lets say the old key was channel-0<>osmosis which points to cosmos. If we want to change it to channel-1<>osmosis points to cosmos, if we don't remove the old key we will have two entries pointing to cosmos AFAI can tell. Lmk if I am misunderstanding.

Reply:
Ahh you're right. It's not possible.